internal/scaffold/role.go

```diff
@@ -143,6 +146,112 @@ func UpdateRoleForResource(r *Resource, absProjectPath string) error {
 	}
 }
 
+// MergeHelmRoleForResource merges incoming new API resource rules with existing deploy/role.yaml
+func MergeHelmRoleForResource(r *Resource, absProjectPath string, roleScaffold Role) error {
+	roleFilePath := filepath.Join(absProjectPath, DeployDir, RoleYamlFile)
+	roleYAML, err := ioutil.ReadFile(roleFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to read role manifest %v: %v", roleFilePath, err)
+	}
+	// Check for existing role.yaml
+	if len(roleYAML) == 0 {
+		return fmt.Errorf("empty Role File at: %v", absProjectPath)
+	}
+	// Check for incoming new Role
+	if len(roleScaffold.CustomRules) == 0 {
+		return fmt.Errorf("customRules cannot be empty for new Role at: %v", r.APIVersion)
+	}
+	if roleScaffold.IsClusterScoped {
+		// TODO: Add logic to merge Cluster scoped rules into existing Kind: Role scoped rules
+		// Check if existing deploy/role.yaml is ClusterScoped already.
+		if !strings.Contains(string(roleYAML), "kind: ClusterRole") {
+			return fmt.Errorf("cannot Merge Cluster scoped rules with existing deploy/role.yaml" +
+				"please modify existing deploy/role.yaml and deploy/role_binding.yaml " +
+				"to reflect Cluster scope and try again")
+		}
+	}
```

Comment:
This check should be moved inside the  `*rbacv1.Role` case block, right? If we have a cluster role already, we can safely merged a role scaffold where IsClusterScoped is true.